starter_project/autonomy/src/perception.cpp

```diff
@@ -71,12 +123,28 @@ namespace mrover {
         // hint: try not overthink, this metric does not have to be perfectly accurate, just correlated to distance away from a tag
 
         // TODO: implement me!
-        return {};
+        //return percent of screen taken off by corners
+        //assume the tag is a square?
+        int total_area = image.rows * image.cols;
+        
+        float corner_area = abs((tagCorners[1].x - tagCorners[0].x) * (tagCorners[0].y - tagCorners[3].y));
+
+        return 1 - float((1.0*corner_area) / (1.0*total_area));
     }
 
-    std::pair<float, float> Perception::getCenterFromTagCorners(std::vector<cv::Point2f> const& tagCorners) { // NOLINT(*-convert-member-functions-to-static)
+    std::pair<float, float> Perception::getCenterFromTagCorners(std::vector<cv::Point2f> const& tagCorners) const { // NOLINT(*-convert-member-functions-to-static)
         // TODO: implement me!
-        return {};
+        float center_x = float(tagCorners[0].x + tagCorners[1].x) / float(2.0);
```

Comment:
You can use `cv::Point2f` from OpenCV which implements element-wise operations. It saves you from defining multiple variables and repetition.